lib/ExportsInfo.js

```diff
@@ -983,11 +1086,106 @@ class ExportInfo {
 		this._usedName = name;
 	}
 
+	/**
+	 * @param {ModuleGraph} moduleGraph the module graph
+	 * @returns {ExportInfo | undefined} the terminal binding export info if known
+	 */
+	getTerminalExportInfo(moduleGraph) {
+		if (this.terminalBinding) return this;
+		const target = this.getTarget(moduleGraph);
+		if (!target) return undefined;
+		const exportsInfo = moduleGraph.getExportsInfo(target.module);
+		return exportsInfo.getReadOnlyExportInfoRecursive(target.export);
+	}
+
+	/**
+	 * @param {ModuleGraph} moduleGraph the module graph
+	 * @param {function({ module: Module, export: string[] | undefined }): boolean} resolveTargetFilter filter function to further resolve target
+	 * @returns {{ module: Module, export: string[] | undefined } | undefined} the target
+	 */
+	getTarget(moduleGraph, resolveTargetFilter = () => true) {
+		return this._getTarget(moduleGraph, resolveTargetFilter);
+	}
+
+	/**
+	 * @param {ModuleGraph} moduleGraph the module graph
+	 * @param {function({ module: Module, export: string[] | undefined }): boolean} resolveTargetFilter filter function to further resolve target
+	 * @param {Set<ExportInfo>} alreadyVisited set of already visited export info to avoid circular references
+	 * @returns {{ module: Module, export: string[] | undefined } | undefined} the target
+	 */
+	_getTarget(
+		moduleGraph,
+		resolveTargetFilter = () => true,
+		alreadyVisited = new Set()
```

Comment:
```suggestion
		alreadyVisited
```
as I see could be initialized with `undefined`